src/sbus/connection/sbus_watch.c

```diff
@@ -280,8 +290,27 @@ sbus_watch_remove(DBusWatch *dbus_watch, void *data)
 
     if (watch_fd->dbus_watch.read == NULL
             && watch_fd->dbus_watch.write == NULL) {
-        talloc_free(watch_fd->fdevent);
-        talloc_free(watch_fd);
+        /* libdbus doesn't need this watch{fd} anymore, so associated
+         * tevent_fd should be removed from monitoring at the spot.
+         *
+         * This first attempt to free will actually fail because tevent_fd is
+         * marked as busy in tevent_common_invoke_fd_handler() but it will
+         * stop fd monitoring in tevent loop.
+         * Memory will be eventually freed alongside with deallocation
+         * of parent context (watch_fd).
+         */
+        TALLOC_FREE(watch_fd->fdevent);
```

Comment:
Is there any specific reason why you used the uppercased macro instead of `talloc_free`? If not, can you change it to lowercase please?

Reply:
`TALLOC_FREE` sets argument to NULL.
I think it is good idea because fdevent will be marked as "destroyed" at this point.

Reply:
Then please use `talloc_zfree` (just to be consistent with the reset of the code) which is does the same thing but is used in sssd. (Historically, it predates TALLOC_ZERO).

Reply:
`TALLOC_FREE` seems safer because it check for (argument != NULL).

But for the sake of consistency... well, ok, I will replace it.